File: aspnet-in-practice/06/VB/CustomControls.Web/CustomControls.ControlBuilders/Parts/CMSBasePart.vb
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Web.UI
Imports System.Web.UI.WebControls

' base class, used to handle the parts in a similar way
Public Class CMSPartBase
	Inherits WebControl
	Protected Sub New()

		MyBase.New()
	End Sub

	' more common properties here - use ViewState if needed
	Public Overridable Property PageSize() As Integer

	Public Property Title() As String
End Class

File: aspnet-in-practice/06/CS/CustomControls.ControlBuilders/Parts/CMSBasePart.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Web.UI;
using System.Web.UI.WebControls;

namespace CustomControls.ControlBuilders
{
	// base class, used to handle the parts in a similar way
	public class CMSPartBase : WebControl
	{
		protected CMSPartBase():base()
		{

		}

		// more common properties here - use ViewState if needed
		public virtual int PageSize { get; set; }
		public string Title { get; set; }
	}
}